perception/autoware_tensorrt_bevdet/src/bevdet.cpp

```diff
@@ -0,0 +1,717 @@
+    {1, {370000}},         {1, {370000}}, {1, {14000}}, {1, {14000}},    {5, {1, 8, 80, 128, 128}},
+    {3, {1, 8, 6}},        {2, {1, 1}}};
+  nvinfer1::IBuilder * builder = nvinfer1::createInferBuilder(g_logger_);
+  nvinfer1::INetworkDefinition * network =
+    builder->createNetworkV2(1U << int(nvinfer1::NetworkDefinitionCreationFlag::kEXPLICIT_BATCH));
+  nvinfer1::IOptimizationProfile * profile = builder->createOptimizationProfile();
+  nvinfer1::IBuilderConfig * config = builder->createBuilderConfig();
+  config->setMemoryPoolLimit(nvinfer1::MemoryPoolType::kWORKSPACE, 3UL << 32UL);
+
+  config->setFlag(nvinfer1::BuilderFlag::kFP16);
+
+  nvonnxparser::IParser * parser = nvonnxparser::createParser(*network, g_logger_);
+
+  if (!parser->parseFromFile(onnxFile.c_str(), static_cast<int>(g_logger_.reportable_severity))) {
+    RCLCPP_ERROR(rclcpp::get_logger("BEVDet"), "Failed parsing .onnx file!");
+    for (int i = 0; i < parser->getNbErrors(); ++i) {
+      auto * error = parser->getError(i);
+      RCLCPP_ERROR(
+        rclcpp::get_logger("BEVDet"), "Error code: %d, Description: %s",
+        static_cast<int>(error->code()), error->desc());
+    }
+
+    return;
+  }
+  RCLCPP_INFO(rclcpp::get_logger("BEVDet"), "Succeeded parsing .onnx file!");
+
+  for (size_t i = 0; i < min_shapes.size(); i++) {
+    nvinfer1::ITensor * it = network->getInput(i);
+    profile->setDimensions(it->getName(), nvinfer1::OptProfileSelector::kMIN, min_shapes[i]);
+    profile->setDimensions(it->getName(), nvinfer1::OptProfileSelector::kOPT, opt_shapes[i]);
+    profile->setDimensions(it->getName(), nvinfer1::OptProfileSelector::kMAX, max_shapes[i]);
+  }
+  config->addOptimizationProfile(profile);
+
+  nvinfer1::IHostMemory * engineString = builder->buildSerializedNetwork(*network, *config);
+  if (engineString == nullptr || engineString->size() == 0) {
+    RCLCPP_ERROR(rclcpp::get_logger("BEVDet"), "Failed building serialized engine!");
+    return;
+  }
+  RCLCPP_INFO(rclcpp::get_logger("BEVDet"), "Succeeded building serialized engine!");
+
+  nvinfer1::IRuntime * runtime{nvinfer1::createInferRuntime(g_logger_)};
+  engine = runtime->deserializeCudaEngine(engineString->data(), engineString->size());
+  if (engine == nullptr) {
+    RCLCPP_ERROR(rclcpp::get_logger("BEVDet"), "Failed building engine!");
+    return;
+  }
+  RCLCPP_INFO(rclcpp::get_logger("BEVDet"), "Succeeded building engine!");
+
+  std::ofstream engineFile(trtFile, std::ios::binary);
+  if (!engineFile) {
+    RCLCPP_ERROR(rclcpp::get_logger("BEVDet"), "Failed opening file to write");
+    return;
+  }
+  engineFile.write(static_cast<char *>(engineString->data()), engineString->size());
+  if (engineFile.fail()) {
+    RCLCPP_ERROR(rclcpp::get_logger("BEVDet"), "Failed saving .engine file!");
+    return;
+  }
+  RCLCPP_INFO(rclcpp::get_logger("BEVDet"), "Succeeded saving .engine file!");
+}
+
+BEVDet::~BEVDet()
```

Comment:
Namespaces: Wrap the destructor `~BEVDet` within the  `autoware_tensorrt_bevdet` namespace.